js/validation.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * input이 공백인지 검사
+ * @param {*} target 이벤트 발생한 타겟 요소
+ * @param {*} errorMessage 에러를 표시할 <span> 태그
+ * @returns input의 입력이 공백이면 true, 공백이 아니면 false 반환
+ */
+function handleEmptyInput(target, errorMessage) {
+  if (target.value.trim("") === "") {
+    errorMessage.textContent = target.placeholder;
+    target.classList.add("invalid");
+    target.dataset.valid = false;
+    return true;
+  } else {
+    target.classList.remove("invalid");
+    target.dataset.valid = true;
+    return false;
+  }
+}
+
+/**
+ * 이메일이 [최소 한글자 이상의 영문 대소문자, 숫자]@[최소 한글자 이상의 영문 대소문자, 숫자].[최소 2글자 이상의 영문 대소문자]의 정규 표현식에 적합한지 검사
+ * @param {*} email 공백을 제거한 이메일 문자열
+ * @returns 이메일이 정규 표현식에 적합하면 true, 적합하지 않으면 false 반환
+ */
+function validateEmailFormat(email) {
+  const emailRegex = /^[a-zA-Z0-9]+@[a-zA-Z0-9]+\.[a-zA-Z]{2,}$/;
+
+  return emailRegex.test(email);
+}
+
+/**
+ * 이메일 입력의 유효성 검사
+ * @param {*} target 이벤트 발생한 타겟 요소
+ * @param {*} errorMessage 에러를 표시할 <span> 태그
+ */
+function validateEmail(target, errorMessage) {
+  const isEmailValid = validateEmailFormat(target.value.trim(""));
+  if (!isEmailValid) {
+    errorMessage.textContent = "잘못된 이메일 형식입니다.";
+    target.classList.add("invalid");
+    target.dataset.valid = false;
+  } else {
+    target.classList.remove("invalid");
+    target.dataset.valid = true;
+  }
+}
+
+/**
+ * 비밀번호 입력의 유효성 검사
+ * @param {*} target 이벤트 발생한 타겟 요소
+ * @param {*} errorMessage 에러를 표시할 <span> 태그
+ */
+function validatePassword(target, errorMessage) {
+  if (target.value.trim("").length < 8) {
+    errorMessage.textContent = "비밀번호를 8자리 이상 입력해주세요.";
+    target.classList.add("invalid");
+    target.dataset.valid = false;
+  } else {
+    target.classList.remove("invalid");
+    target.dataset.valid = true;
+  }
+}
+
+/**
+ * 비밀번호 확인 입력의 유효성 검사
+ * @param {*} target 이벤트 발생한 타겟 요소
+ * @param {*} errorMessage 에러를 표시할 <span> 태그
+ */
+function validatePasswordConfirm(target, errorMessage) {
+  const password = target
+    .closest("form")
+    .querySelector("#input_password")
+    .value.trim("");
+
+  if (target.value.trim("") !== password) {
+    errorMessage.textContent = "비밀번호가 일치하지 않습니다.";
+    target.classList.add("invalid");
+    target.dataset.valid = false;
+  } else {
+    validatePassword(target, errorMessage);
+  }
+}
+
+/**
+ * form의 input 요소들의 유효성 검사
+ */
+function checkInputValidity({ target }) {
+  const errorMessage = target
+    .closest(".input-area")
+    .querySelector(".msg-error");
+  const isEmpty = handleEmptyInput(target, errorMessage);
+
+  if (!isEmpty) {
```

Comment:
if 문 안에 switch 문을 중첩해서 작성하는건 가독성이 좋지 않아질 수 있습니다. 아래와 같이 수정하면 좋을 것 같네요!
```suggestion

 // isEmpty === true 인 경우 함수 종료
  if (isEmpty) {
  return;
  }
  
  // 그 외의 경우에만 switch 문 돌림
  switch (targe.id) {
  ...
  }
```